extendr-api/tests/ndarray_try_from_tests.rs

```diff
@@ -0,0 +1,347 @@
+#[cfg(feature = "ndarray")]
+mod ndarray_try_from_tests {
+    use extendr_api::prelude::*;
+
+    #[test]
+    fn integer_matrix_2d() {
```

Comment:
These tests all seem to have the same structure. Could you reduce redundancy using a `macro_rules!`?

Reply:
Honestly, no idea how to do that, since every case is slightly different from another. Test cases are pretty much hand-picked. Maybe [this](https://docs.rs/rstest/latest/rstest/) can help, but I never used it & I'd prefer to handle it in a separate PR (to unblock progress for others).